src/mam/mod_mam_riak_timed_arch_yz.erl

```diff
@@ -132,10 +132,10 @@ archive_message(_Result, Host, MessId, _UserID, LocJID, RemJID, SrcJID, _Dir, Pa
             {error, Reason}
     end.
 
-archive_message_muc(_Result, Host, MessId, _UserID, LocJID, _FromJID, SrcJID, _Dir, Packet) ->
-    RemJIDMuc = maybe_muc_jid(SrcJID),
+archive_message_muc(_Result, Host, MessId, _UserID, LocJID, FromJID, SrcJID, _Dir, Packet) ->
+    BareFromJID = jid:to_bare(FromJID),
     try
-        archive_message(Host, MessId, LocJID, RemJIDMuc, SrcJID, Packet, muc)
+        archive_message(Host, MessId, LocJID, BareFromJID, SrcJID, Packet, muc)
```

Comment:
This is the real schema change:
`LocJID` is the room's JID, i.e., 'room@domain'
`FromJID` is the real user's JID, i.e., 'user@domain/resource'
`SrcJID` is the user's JID in the room, i.e., 'room@domain/nickname'
If we do `maybe_muc_jid(SrcJID)`, of something with a resource, we will archive by this resource. But we want to archive by the user, so we can query so later. Unfortunately this makes it archive by the same pattern than private messages, so we need to filter later by type, as we do in this PR.